test/remote/gateways/remote_pay_trace_test.rb

```diff
@@ -0,0 +1,201 @@
+require 'test_helper'
+
+class RemotePayTraceTest < Test::Unit::TestCase
+  def setup
+    @gateway = PayTraceGateway.new(fixtures(:pay_trace))
+
+    @amount = 100
+    @credit_card = credit_card('4012000098765439')
+    @mastercard = credit_card('5499740000000057')
+    @discover = credit_card('6011000993026909')
+    @amex = credit_card('371449635392376')
+    @options = {
+      billing_address: {
+        address1: '8320 This Way Lane',
+        city: 'Placeville',
+        state: 'CA',
+        zip: '85284'
+      },
+      description: 'Store Purchase'
+    }
+  end
+
+  def test_acquire_token
+    response = @gateway.acquire_access_token
+    assert_not_nil response['access_token']
+  end
+
+  def test_successful_purchase
+    response = @gateway.purchase(@amount, @credit_card, @options)
+    assert_success response
+    assert_equal 'Your transaction was successfully approved.', response.message
+  end
+
+  def test_successful_purchase_with_customer_id
+    create = @gateway.store(@mastercard, @options)
+    customer_id = create.params['customer_id']
+    response = @gateway.purchase(500, customer_id, @options)
+    assert_success response
+    assert_equal 'Your transaction was successfully approved.', response.message
+  end
+
+  def test_successful_purchase_with_more_options
+    options = {
+      email: 'joe@example.com'
+    }
+
+    response = @gateway.purchase(200, @discover, options)
+    assert_success response
+    assert_equal 'Your transaction was successfully approved.', response.message
+  end
+
+  def test_failed_purchase
+    response = @gateway.purchase(29, @amex, @options)
+    assert_failure response
+    assert_equal false, response.success?
+  end
+
+  def test_successful_authorize_and_capture
+    auth = @gateway.authorize(300, @credit_card, @options)
+    assert_success auth
+
+    assert capture = @gateway.capture(auth.authorization, @options.merge(amount: 3.00))
```

Comment:
woaaah why is this a decimal when the rest are not?

Reply:
Great question! Because it is a parameter being given in the options hash instead of as a 'standalone'. Or at least that is what I have discovered to be the case.

Reply:
I think we maybe should be passing this option through the [`amount` method](https://github.com/activemerchant/active_merchant/blob/a9c588f90c4c1d59f7b0f5e1dc40b9ca12cb7fbf/lib/active_merchant/billing/gateway.rb#L250) so that it can obey the `money_format`, the way we're already doing in `add_invoice`?